app/controllers/topics_controller.rb

```diff
@@ -288,6 +288,14 @@ def update
       else
         return render_json_error(I18n.t('category.errors.not_found'))
       end
+
+      if category && topic_tags = params[:tags] || topic.tags.pluck(:name)
+        allowed_tags = category.tags.pluck(:name)
+
+        if !(topic_tags.blank? || allowed_tags.blank?) && !(topic_tags - allowed_tags).empty?
```

Comment:
Since you're performing the set difference twice `topic_tags - allowed_tags`) it would be more efficient to store the result in a variable.